### DISCUSSION TITLE
How to push new version to PyPi

### DISCUSSION THREAD

**Discussion by XenonPy** [mojidev-py/discmoji#16] [General]
like how to increment patch, major, minor, etc. and push it to GH releases and Pip

**Comment by mojidev-py** [comment]
hmmm first create a pypi account, and I'll add you to the project.
## STEPS ON HOW TO INCREMENT VERSION IN PyPi:
1. First go to `pyproject.toml`, and change the version by one (really important, this is how the versions are changed)
2. do `python -m build`
3. Once you've done that, install twine if you haven't, and type in `twine upload dist/*`
4. It'll ask you for your API token, which can be recieved through your setting dashboard in PyPi
5. Once you've entered that, it should be uploaded to PyPi, and pip automatically searches PyPi for the package, which means if it's up to date on PyPi, it's up to date on pip
GH releases, just create a release through the repo's release section, if you don't have the perms, then just open a discussion

**Comment by mojidev-py** [comment]
python -m build should create a folder called `discmoji-egg.info`

**Comment by XenonPy** [comment]
ok I will do this in like 1.5 hrs

**Comment by mojidev-py** [comment]
btw, you can edit the todo.md if you finished trying to implement something

**Reply by XenonPy** [reply]
ok thanks

**Reply by mojidev-py** [reply]
btw, what's ur PyPi username?
